crates/vm/levm/src/vm.rs

```diff
@@ -1303,6 +1371,226 @@ impl VM {
 
         Ok(report)
     }
+
+    pub fn eip7702_set_access_code(
+        &mut self,
+        initial_call_frame: &mut CallFrame,
+    ) -> Result<u64, VMError> {
+        let mut refunded_gas: u64 = 0;
+        // Steps from the EIP7702:
+        // IMPORTANT:
+        // If any of the below steps fail, immediately stop processing that tuple and continue to the next tuple in the list. It will in the case of multiple tuples for the same authority, set the code using the address in the last valid occurrence.
+        // If transaction execution results in failure (any exceptional condition or code reverting), setting delegation designations is not rolled back.
+        // TODO: avoid clone()
+        for auth_tuple in self.authorization_list.clone().unwrap_or_default() {
+            let chain_id_not_equals_this_chain_id = auth_tuple.chain_id != self.env.chain_id;
+            let chain_id_not_zero = !auth_tuple.chain_id.is_zero();
+
+            // 1. Verify the chain id is either 0 or the chain’s current ID.
+            if chain_id_not_zero && chain_id_not_equals_this_chain_id {
+                continue;
+            }
+
+            // 2. Verify the nonce is less than 2**64 - 1.
+            // NOTE: nonce is a u64, it's always less or equal to u64::MAX
+            if auth_tuple.nonce == u64::MAX {
+                continue;
+            }
+
+            // 3. authority = ecrecover(keccak(MAGIC || rlp([chain_id, address, nonce])), y_parity, r, s)
+            //      s value must be less than or equal to secp256k1n/2, as specified in EIP-2.
```

Comment:
It'd be nice to have something like this here.
```suggestion
            // 3. authority = ecrecover(keccak(MAGIC || rlp([chain_id, address, nonce])), y_parity, r, s)
            //      s value must be less than or equal to secp256k1n/2, as specified in EIP-2.
            let Some(authority) = xxx(...) else {
                continue;
            }
```

Reply:
working on it

Reply:
Done in commit [f0c3d7d](https://github.com/lambdaclass/ethrex/pull/1672/commits/f0c3d7d6a7afcd8fa54184d4068933bb3f51127b)